tck/src/main/java/com/akkaserverless/javasdk/tck/model/localpersistenceeventing/LocalPersistenceSubscriber.java

```diff
@@ -18,63 +18,73 @@
 
 import akka.NotUsed;
 import akka.stream.javadsl.Source;
-import com.akkaserverless.javasdk.CloudEvent;
 import com.akkaserverless.javasdk.action.Action;
 import com.akkaserverless.javasdk.action.ActionContext;
 import com.akkaserverless.javasdk.action.ActionCreationContext;
 import com.akkaserverless.tck.model.eventing.LocalPersistenceSubscriberModel;
 import com.akkaserverless.tck.model.eventing.LocalPersistenceEventing;
+import com.google.protobuf.Any;
 
 public class LocalPersistenceSubscriber extends Action {
 
   public LocalPersistenceSubscriber(ActionCreationContext creationContext) {}
 
   public Action.Effect<LocalPersistenceEventing.Response> processEventOne(
-      ActionContext context, CloudEvent cloudEvent, LocalPersistenceEventing.EventOne eventOne) {
-    return convert(context, cloudEvent, eventOne.getStep());
+      LocalPersistenceEventing.EventOne eventOne) {
+    return convert(actionContext(), eventOne.getStep());
   }
 
   public Source<Action.Effect<LocalPersistenceEventing.Response>, NotUsed> processEventTwo(
-      ActionContext context, CloudEvent cloudEvent, LocalPersistenceEventing.EventTwo eventTwo) {
-    return Source.from(eventTwo.getStepList()).map(step -> convert(context, cloudEvent, step));
+      LocalPersistenceEventing.EventTwo eventTwo) {
+    ActionContext context = actionContext();
+    return Source.from(eventTwo.getStepList()).map(step -> convert(context, step));
   }
 
-  public LocalPersistenceEventing.Response processAnyEvent(
-      JsonMessage jsonMessage, CloudEvent cloudEvent) {
-    return LocalPersistenceEventing.Response.newBuilder()
-        .setId(cloudEvent.subject().orElse(""))
-        .setMessage(jsonMessage.message)
-        .build();
+  public Action.Effect<LocalPersistenceEventing.Response> processAnyEvent(Any any) {
+    JsonMessage jsonMessage = actionContext().deserializeJson(JsonMessage.class, any);
```

Comment:
I like this explicit approach but I don't know how it was intended to work before and if there was some strong opinions about that it should be automagic and not be part of user's business logic code.

Reply:
The alternative as I see it is with reflection out of the picture is to introduce protobuf annotations describing the possible Java classes that the json could be deserialized to and then generate methods for each of those types as well as logic to try to deserialize the right type in the handler. That would add more complexity here, require bigger changes all the way back to the protocol/annotations adding another case of stringly typed references to the Java codebase in the protobuf.

Reply:
Unless I miss something we should focus on the json support for pub-sub. Since that is done via actions and the main responsibility of such actions is to translate between internal/external protocols I think it's a good fit to handle json there as well. Explicitly.

Reply:
Ok, I may be missing the whole json support story here, but it isn't clear to me how that will look like for a generated action. The problem with this example from the TCK is that it's fully hand-coded. 

For illustration, let's take the following protobuf fragment

```
rpc SayHello(Person) returns (Hello) {}
```

this will generate a method as

```java
public Action.Effect<Hello> sayHello(Person person)
```

How would we know that when generating this Action, we will need to pass a String or Any and that the user will have to manually deserialize it to the right type? 

So, the only way out of this to me is to say, if you want to pass json to an action, you have to encode it by hand yourself, but that means that user will have to generate a Provider and a Handler. And that bring us to the next question, how we will tell Akka Serverless how such an Action needs to be exposed? What are the endpoints? 

It seems to me that whatever we do, a proto service definition is always needed and it's in the proto file that we will need to tell if the types will be ultimately serialized to json.

Reply:
So if a user is building an action that subscribes to a topic, they will need to _know_ what serialization format the messages are, and type their action input parameter as `google.protobuf.Any` (just like how it is now), then do an explicit deserialization to their expected internal type. We provide Jackson parsing json support out of the box to deserialize to pojos since that might be common enough.

If a user is publishing to a topic and do not want to use protobuf they will instead use `google.protobuf.Any` as return value, and have explicit logic to turn their object into the right serialized format. Again we provide explicit support to turn a pojo into json and then that into a `google.protobuf.Any` with the akka serverless json type-url.

If a user wants to produce or consume some other event encoding with an arbitrary third party library for serialization/deserialization they have access to the raw bytes from the `google.protobuf.Any` and can do that (not sure about how/if the proxy deals with that currently though).

Reply:
The explicit deserialisation looks good to me too. And it feels like it will be clearer that the protobuf rpc method and the action implementation method both take Any.